### PR TITLE
Keychron q5

### PR DESCRIPTION
**Pull request by lalalademaxiya1**
Add new keyboard q5 which contains four sub versions.

## Description

- Add keyboard rev_0150, ansi version of q5.
- Add keyboard rev_0151, ansi version of q5 and enable encoder ec11.
- Add keyboard rev_0152, iso version of q5.
- Add keyboard rev_0153, iso version of q5 and enable encoder ec11.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by zvecr — CHANGES_REQUESTED** [review]
`ENCODER_MAP_ENABLE` should only be set at the keymap level

File: keyboards/keychron/q5/matrix.c [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+#endif // MATRIX_ROW_PINS
+#ifdef MATRIX_COL_PINS
+static pin_t col_pins[MATRIX_COLS] = MATRIX_COL_PINS;
+#endif // MATRIX_COL_PINS
+
+#define ROWS_PER_HAND (MATRIX_ROWS)
+
+/* matrix state(1:on, 0:off) */
+extern matrix_row_t raw_matrix[MATRIX_ROWS]; // raw values
+extern matrix_row_t matrix[MATRIX_ROWS];     // debounced values
```
Comment (zvecr) [inline keyboards/keychron/q5/matrix.c:45]:
```suggestion
```

File: keyboards/keychron/q5/halconf.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define HAL_USE_I2C TRUE
+#define PAL_USE_CALLBACKS TRUE
```
Comment (zvecr) [inline keyboards/keychron/q5/halconf.h:20]:
```suggestion
```

File: keyboards/keychron/q5/matrix.c [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    } else {
+        shiftOut(0xFF);
+    }
+}
+
+// static void unselect_cols(void) {
+//     for (uint8_t x = 0; x < MATRIX_COLS; x++) {
+//         unselect_col(x);
+//     }
+// }
```
Comment (zvecr) [inline keyboards/keychron/q5/matrix.c:119]:
```suggestion
```

File: keyboards/keychron/q5/readme.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+
+    make keychron/q5/rev_0150:default
+
+Flashing example for this keyboard:
+
+    make keychron/q5/rev_0150:default:flash
+
+**Reset Key**: Hold down the key located at *K00*, commonly programmed as *Esc* while plugging in the keyboard.
```
Comment (zvecr) [inline keyboards/keychron/q5/readme.md:17]: Probably want to change this to maybe describe what each revision is, and to direct users to the revision readme for more info.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q5/readme.md (hunk 4 above)
Comment (lalalademaxiya1) [reply keyboards/keychron/q5/readme.md]: Thanks, i will arrange it.

File: keyboards/keychron/q5/q5.c [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                    } break;
+                }
+            }
+            return false;
+#    endif
+    }
+    return true;
+}
+
+__attribute__((weak)) void rgb_matrix_indicators_advanced_kb(uint8_t led_min, uint8_t led_max) {
```
Comment (drashna) [inline keyboards/keychron/q5/q5.c:68]:
```suggestion
__attribute__((weak)) void rgb_matrix_indicators_advanced_user(uint8_t led_min, uint8_t led_max) {
```

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by keychron-dev** [comment]
hi, please help check this. thanks

**Comment by keychron-dev** [comment]
@zvecr do you have time to check this one? thanks

**Pull request merged**